src/components/TableWithInlineEditing.js

```diff
@@ -130,67 +168,75 @@ const Row = ({
-                                                ) : (
-                                                    <InputField
-                                                        type={column.type}
-                                                        value={rowContent[column.key]}
-                                                        onChange={handleCellChange}
-                                                        propertyName={column.key}
-                                                    />
-                                                )}
-                                            </div>
-                                        )}
-                                    </TableCell>
-                                </Fragment>
-                            ))}
-                        </Fragment>
-                    ) : (
-                        //readonly for users without edit permission
-                        <Fragment>
-                            {columnsInfo.map(column => (
-                                <Fragment key={column.title}>
-                                    <TableCell>
-                                        <span name={column.key} className={classes.notClickable}>
+        <TableRow key={rowContent.id}>
+            {rowContent !== {} &&
+                columnsInfo &&
+                (allowedToEdit ? (
+                    <Fragment>
+                        {columnsInfo.map((column, index) => (
+                            <Fragment key={column.title}>
+                                <TableCell
+                                    onClick={() => changeCell(`${rowIndex}${column.key}`)}
+                                    onKeyDown={e => handleKeyPress(e, index)}
+                                    className={classes.pointer}
+                                >
+                                    {currentlyEditing !== `${rowIndex}${column.key}` ? (
+                                        <span id={`outer${rowIndex}-${index}`} name={column.key}>
                                             {rowContent[column.key]}
                                         </span>
-                                    </TableCell>
-                                </Fragment>
-                            ))}
-                        </Fragment>
-                    ))}
-            </TableRow>
-        </Fragment>
+                                    ) : (
+                                        <div id={`inner${rowIndex}-${index}`}>
+                                            {column.type === 'dropdown' ? (
+                                                <Dropdown
+                                                    onChange={handleCellChange}
+                                                    items={column.options}
+                                                    value={rowContent[column.key]}
+                                                    propertyName={column.key}
```

Comment:
Not really a change request just an observation.... Dropdown takes a new optional 'allowNoValue' prop here which defaults to true. In its current form, every dropdown the table ever renders will allow a blank value. This seems reasonable to me but not sure if your use case requires restricting of blank values? If it does you might need to handle this at your component level.

Reply:
hmm, I think I might pass in false from this component for now as I don't think I'd want blank values in my own use case, then if we need the option in future we can change the table component to take in whether it allows blank for dropdowns